Tests/MongoSwiftTests/AsyncAwaitTests.swift

```diff
@@ -123,21 +44,74 @@ final class AsyncAwaitTests: MongoSwiftTestCase {
                 ) {
                     client.connectionPool.checkedOutConnections == 0
                 }
+            }
+        }
+    }
+
+    func testTransactions() throws {
+        testAsync {
+            try await self.withTestNamespace { client, _, coll in
+                guard try await client.supportsTransactions() else {
+                    printSkipMessage(testName: self.name, reason: "Requires transactions support")
+                    return
+                }
+
+                // aborted txn
+                try await client.withSession { session in
+                    try await session.startTransaction()
+                    try await coll.insertOne(["_id": 1], session: session)
+                    try await session.abortTransaction()
+
+                    let count = try await coll.countDocuments(session: session)
+                    expect(count).to(equal(0))
+
+                    let doc = try await coll.findOne(session: session)
+                    expect(doc).to(beNil())
+                }
 
-                // TODO: SWIFT-1391 once we have more API methods available, test transaction usage here.
+                // committed txn
+                try await client.withSession { session in
+                    try await session.startTransaction()
+                    try await coll.insertOne(["_id": 1], session: session)
+                    try await session.commitTransaction()
+
+                    let count = try await coll.countDocuments(session: session)
+                    expect(count).to(equal(1))
+
+                    let doc = try await coll.findOne(session: session)
+                    expect(doc).to(equal(["_id": 1]))
+                }
             }
         }
     }
 
     func testMongoDatabase() throws {
         testAsync {
             try await self.withTestNamespace { _, db, _ in
+                try await db.drop()
```

Comment:
I think I didn't notice this issue with my last PR because when you run the whole suite the test DB ends up getting cleaned up somewhere else, but running just the async/await tests multiple times in a row leads to failure here since test.foo wasn't being cleaned up.